src/main/java/org/anarres/parallelgzip/ParallelGZIPOutputStream.java

```diff
@@ -189,7 +202,8 @@ public void write(byte[] b, int off, int len) throws IOException {
     private void submit() throws IOException {
         emitUntil(nthreads - 1);
         emitQueue.add(executor.submit(block));
-        block = new Block(/* block.index + 1 */);
+        block = blocks.isEmpty() ? new Block() : blocks.remove(blocks.size() - 1);
```

Comment:
Edited in memory-allocation-improvements - same as ArrayList really, but a cleaner API for removeLast().